pkg/controller/master/image/backing_image_controller.go

```diff
@@ -80,6 +104,9 @@ func (h *backingImageHandler) OnChanged(_ string, backingImage *lhv1beta2.Backin
 			toUpdate.Status.Progress = status.Progress
 			toUpdate.Status.Size = backingImage.Status.Size
 			toUpdate.Status.VirtualSize = backingImage.Status.VirtualSize
+			toUpdate.Status.Failed = 0
```

Comment:
This might have a potential issue, and could make VirtualMachineImage trigger `handleRetry` forever, right ? Assume we have 3 disk files in backingimage, first in failed state, second & third in ready state. And each time we got the same DiskFileStatusMap (i.e. 1 failed, 2 ready), then VirtualMachineImage retry never ends as the failed count is reset to 0 every time, and controller keeps delete BackingImage and StorageClass in each retry.

I'm not sure if this is something we need to aware of, if that's not the case, please ignore this comment.

Reply:
https://github.com/harvester/harvester/pull/6915#discussion_r1827182479
As you mentioned, we should not reset the imported condition, otherwise, it will make the image controller delete/recreate the backing image and storage class, thanks for pointing this.